multimodal_knowledge_retriever/src/multimodal.py

```diff
@@ -301,20 +301,51 @@ def vectorstore_ingest(self, retriever, text_docs, table_docs, image_paths, summ
         
         return retriever
     
-    def set_retrieval_chain(self, retriever):
+    def get_retrieved_images_and_docs(self, retriever, query):
+        results=retriever.invoke(query)
+        image_results = [result for result in results if result.metadata["type"]=="image"]
+        doc_results = [result for result in results if result.metadata["type"]!="image"]
+        return image_results, doc_results
+    
+    def get_image_answers(self, retrieved_image_docs, query):
+        image_answer_prompt_template = load_prompt(os.path.join(kit_dir,"prompts","llava-qa.yaml"))
+        image_answer_prompt = image_answer_prompt_template.format(question = query)
+        answers = []
+        for doc in retrieved_image_docs:
+            image_path = os.path.join(doc.metadata["file_directory"],doc.metadata["filename"])
+            answers.append(self.llava_call(image_answer_prompt, image_path))
+        return answers
+    
+    def set_retrieval_chain(self, retriever, image_retrieval_type="raw"):
```

Comment:
Is this intentionally set instead of get?

Reply:
Rigth should be  get, changed in next commit